exec/plugin-commands-script-runners/src/runDepsStatusCheck.ts

```diff
@@ -0,0 +1,39 @@
+import { type VerifyDepsBeforeRun } from '@pnpm/config'
+import { PnpmError } from '@pnpm/error'
+import { globalWarn } from '@pnpm/logger'
+import { install } from '@pnpm/plugin-commands-installation'
+import { checkDepsStatus, type CheckDepsStatusOptions } from '@pnpm/deps.status'
+import { prompt } from 'enquirer'
+
+export type RunDepsStatusCheckOptions = CheckDepsStatusOptions & { verifyDepsBeforeRun?: VerifyDepsBeforeRun }
+
+export async function runDepsStatusCheck (opts: RunDepsStatusCheckOptions): Promise<void> {
+  const { upToDate, issue } = await checkDepsStatus(opts)
+  if (upToDate) return
+  switch (opts.verifyDepsBeforeRun) {
+  case 'install':
+    await install.handler(opts as unknown as install.InstallCommandOptions)
+    break
+  case 'prompt': {
+    const confirmed = await prompt<{ runInstall: boolean }>({
+      type: 'confirm',
+      name: 'runInstall',
+      message: `Your "node_modules" directory is out of sync with the "pnpm-lock.yaml" file. This can lead to issues during scripts execution.
+
+Would you like to run "pnpm install" to update your "node_modules"?`,
+      initial: true,
+    })
+    if (confirmed.runInstall) {
```

Comment:
Not a blocker, just a question that probably need to be documented: Does confirming this means that the installation potentially updates the lockfile?

Reply:
Yes, I think that could change the lockfile if needed.